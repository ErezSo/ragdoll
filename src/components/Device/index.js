import React from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  border: 1px solid black;
  padding-right: 38px;
  margin: 20px;
`;
const List = styled.ul`
  list-style-type: "none";
`;

const Device = ({ device, toggleDevice, disableButton }) => {
  return (
    <Wrapper>
      <List>
        <li>Name: {device.name}</li>
        <li>Unit: {device.unit}</li>
        <li>Value: {device.value}</li>
        <li>Timestamp: {new Date(device.timestamp).toLocaleString()}</li>
        <li>
          Status: {device.active ? "Active" : "Inactive"}{" "}
          <span>
            <button
              disabled={disableButton}
              onClick={() => toggleDevice(device)}
            >
              Toggle
            </button>
          </span>
        </li>
      </List>
    </Wrapper>
  );
};

export default Device;
