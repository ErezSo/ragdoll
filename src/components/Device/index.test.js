import React from "react";
import { shallow } from "enzyme";
import Device from "./";

describe("Device", () => {
  let wrapper;
  const props = {
    device: {
      name: "rotation_rate_gamma",
      unit: "deg/s",
      value: 0,
      timestamp: 1528878066506,
      active: true
    },
    toggleDevice: jest.fn(),
    disableButton: true
  };
  beforeEach(() => {
    wrapper = shallow(<Device {...props} />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
