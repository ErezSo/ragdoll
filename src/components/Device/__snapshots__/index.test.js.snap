// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device renders correctly 1`] = `
"<div style={{...}}>
  <ul style={{...}}>
    <li>
      Name: 
      rotation_rate_gamma
    </li>
    <li>
      Unit: 
      deg/s
    </li>
    <li>
      Value: 
      0
    </li>
    <li>
      Timestamp: 
      2018-6-13 10:21:06
    </li>
    <li>
      Status: 
      Active
       
      <span>
        <button disabled={true} onClick={[Function: onClick]}>
          Toggle
        </button>
      </span>
    </li>
  </ul>
</div>"
`;
